lte/gateway/python/magma/mobilityd/ip_address_man.py

```diff
@@ -132,25 +135,23 @@ def __init__(self,
         self._recycling_interval_seconds = recycling_interval
 
         if not persist_to_redis:
+            client = None
             self._assigned_ip_blocks = set()  # {ip_block}
             self.sid_ips_map = defaultdict(IPDesc)  # {SID=>IPDesc}
         else:
-            if not redis_port:
-                raise ValueError(
-                    'Must specify a redis_port in mobilityd config.')
-            client = get_default_client()
+            client = redis.Redis(host=redis_host, port=redis_port)
             self._assigned_ip_blocks = store.AssignedIpBlocksSet(client)
             self.sid_ips_map = store.IPDescDict(client)
 
-        self.ip_state_map = IpDescriptorMap(persist_to_redis, redis_port)
+        self.ip_state_map = IpDescriptorMap(persist_to_redis, redis_port, redis_host)
         logging.info("Using allocator: %s", self.allocator_type)
         if self.allocator_type == MobilityD.IP_POOL:
             self.ip_allocator = IpAllocatorStatic(self._assigned_ip_blocks,
                                                   self.ip_state_map,
                                                   self.sid_ips_map)
         elif self.allocator_type == MobilityD.DHCP:
-            dhcp_store = store.MacToIP()  # mac => DHCP_State
-            dhcp_gw_info = UplinkGatewayInfo(store.GatewayInfoMap())
+            dhcp_store = store.MacToIP(client)  # mac => DHCP_State
```

Comment:
Above, we set client to None if redis persistence is disabled. Does this mean that DHCP IP allocation just doesn't work without stateless mode? Will it crash the mobilityd service?

Reply:
I think we should be instantiating the `dhcp_store` and `dhcp_gw_info` as plain dict() if redis persistence is disabled